pinot-spi/src/main/java/org/apache/pinot/spi/stream/StreamMetadataProvider.java

```diff
@@ -86,7 +87,24 @@ StreamPartitionMsgOffset fetchStreamPartitionOffset(OffsetCriteria offsetCriteri
   default List<PartitionGroupMetadata> computePartitionGroupMetadata(String clientId, StreamConfig streamConfig,
       List<PartitionGroupConsumptionStatus> partitionGroupConsumptionStatuses, int timeoutMillis)
       throws IOException, TimeoutException {
-    int partitionCount = fetchPartitionCount(timeoutMillis);
+
+    int partitionCount;
+    try {
+      partitionCount = fetchPartitionCount(timeoutMillis);
+    } catch (Exception e) {
+      LOGGER.warn("Failed to fetch partition count for stream config: {}. Skipping stream and using"
+      + "existing partitions only. Error: {}", streamConfig.getTopicName(), e.getMessage(), e);
+      // Return only the existing partition groups if we can't fetch partition count
+      List<PartitionGroupMetadata> existingPartitionGroupMetadataList =
+      new ArrayList<>(partitionGroupConsumptionStatuses.size());
+      for (PartitionGroupConsumptionStatus currentPartitionGroupConsumptionStatus : partitionGroupConsumptionStatuses) {
+        existingPartitionGroupMetadataList.add(
+            new PartitionGroupMetadata(currentPartitionGroupConsumptionStatus.getStreamPartitionGroupId(),
+                currentPartitionGroupConsumptionStatus.getEndOffset()));
```

Comment:
How is this `partitionGroupConsumptionStatuses` updated?

Is it possible this `endOffset` is lower than the previous commit segment end offset?